src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -925,40 +934,47 @@ private SymbolAndDiagnostics ComputeImplementationAndDiagnosticsForInterfaceMemb
 
 #if !DEBUG
             // Don't optimize in DEBUG for better coverage for the GetInterfaceLocation function. 
-            if (useSiteDiagnostics != null)
+            if (useSiteDiagnostics != null && implementingTypeImplementsInterface)
```

Comment:
Does this make it so we add the `useSiteDiagnostics` in a release build even if `implementingTypeImplementsInterface` is `false`?

Reply:
> however it still seems like there might be a behavior inconsistency between debug and release builds.

That behavior difference is not observable, unless there is a bug in the method within the ```if``` statement. We intentionally stress them in DEBUG. There are other examples where we do extra work in DEBUG for the sake of extra validation.

---
In reply to: [393223483](https://github.com/dotnet/roslyn/pull/42405#discussion_r393223483) [](ancestors = 393223483)